src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -91,11 +91,43 @@ else if ( alwaysGenerateASandXS ) {
     }
 
     /**
+     * Produces an SA tag for each primary line and supplemental alignment as a map from alignment to tag value.
+     */
+    public static Map<BwaMemAlignment,String> createSATags( final List<BwaMemAlignment> alignments,
+                                                            final List<String> refNames ) {
+        final int nAlignments = alignments.size();
+        if ( nAlignments < 2 ) return Collections.emptyMap();
+
+        final String[] selfTags = new String[nAlignments];
+        for ( int idx = 0; idx != nAlignments; ++idx ) {
+            final BwaMemAlignment alignment = alignments.get(idx);
+            if ( SAMFlag.NOT_PRIMARY_ALIGNMENT.isUnset(alignment.getSamFlag()) ) {
+                selfTags[idx] = asTag(alignment, refNames);
+            }
+        }
+
+        final Map<BwaMemAlignment,String> saTags = new HashMap<>(SVUtils.hashMapCapacity(nAlignments));
+        for ( int idx = 0; idx != nAlignments; ++idx ) {
+            final BwaMemAlignment alignment = alignments.get(idx);
+            if ( SAMFlag.NOT_PRIMARY_ALIGNMENT.isSet(alignment.getSamFlag()) ) continue;
+
+            final StringBuilder saTag = new StringBuilder();
+            for ( int idx2 = 0; idx2 != nAlignments; ++idx2 ) {
+                if ( idx2 == idx ) continue;
+                final String tag = selfTags[idx2];
+                if ( tag != null ) saTag.append(tag);
+            }
+            saTags.put(alignment, saTag.toString());
+        }
+        return saTags;
+    }
+
+    /**
      * Describes an alignment as a string for use in an SA tag, for example.
      */
     public static String asTag( final BwaMemAlignment alignment, final List<String> refNames ) {
```

Comment:
This method seems to deal with SA tag only and is used only by `createSATags `. I would suggest making it private, and rename to something like `asSATag` (doesn't have to be, just for the name to reflect the intention a bit better, or the XA tag could be formatted like this as well?)